feature: add sqlite support

muco-rolle opened the pull request:
<!---
Please carefully read the contribution docs before creating a pull request
 👉 https://github.com/adonisjs/.github/blob/main/docs/CONTRIBUTING.md
-->

<!-- Please ensure there is an open issue and mention its number as #123 -->

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [x] ✨ New feature (a non-breaking change that adds functionality)


### 📚 Description
 
This PR upgrades `node-rate-limiter-flexible` to the latest version, which includes SQLite support. It also updates the store configuration to integrate SQLite as a supported option.  

#### Background  
Our project uses SQLite as the main database, and some features require rate limiting. To enable this, I contributed SQLite support to `node-rate-limiter-flexible`. With this update, we can now leverage SQLite as a storage backend for rate limiting within this Adonis package.  

### Changes  
- Upgraded `node-rate-limiter-flexible` to the latest version  
- Updated the  rate limiter store to support SQLite  

### Testing  
- Verified that the rate limiter works correctly with SQLite as the storage backend  
- Ensured that existing functionality remains unaffected  


Let me know if any adjustments or improvements are needed!

muco-rolle committed:
feature: add sqlite support

muco-rolle commented:
Hello @thetutlage, could you please take a look at this?

thetutlage commented on package.json:
```diff
@@ -20,6 +20,7 @@
     "test": "npm run test:pg && npm run test:mysql && mkdir coverage/tmp && cp -r coverage/*/tmp/. coverage/tmp && c8 report",
     "test:pg": "cross-env DB=pg c8 --reporter=json --report-dir=coverage/pg npm run quick:test",
     "test:mysql": "cross-env DB=mysql c8 --reporter=json --report-dir=coverage/mysql npm run quick:test",
+    "test:sqlite": "cross-env DB=sqlite c8 --reporter=json --report-dir=coverage/sqlite npm run quick:test",
```
This script should be added to the Github workflows for tests to run for SQLite

thetutlage commented:
Can you please fix the type-checking errors? https://github.com/adonisjs/limiter/pull/14/files#file-tests-define_config-spec-ts-L166

muco-rolle committed:
chore(ci): add SQLite tests job to GitHub Actions workflow

muco-rolle committed:
fix(test): update type test to fix parameter type mismatch

muco-rolle commented:
@thetutlage Hi,

I've addressed the comments.

However, it looks like the `toMatchTypeOf` matcher is deprecated. Do you think we should remove it?

thetutlage commented:
I see the SQLite tests are failing. Can you please look into them?

thetutlage commented:
> However, it looks like the toMatchTypeOf matcher is deprecated

We can replace it with the `toEqualTypeOf`

muco-rolle committed:
refactor(tests): replace deprecated `toMatchTypeOf` matcher with `toEqualTypeOf`

muco-rolle committed:
chore(ci): add redis service to test-sqlite job

muco-rolle commented:
> I see the SQLite tests are failing. Can you please look into them?

Hello @thetutlage 

Redis was missing in the test-sqlite job, which seemed to be the reason the tests were failing. I've added it now, and it should be resolved. Let me know if everything looks good.

thetutlage commented:
Thanks for the PR @muco-rolle 🚀
